src/components/modals/ScreenshotModal/ScreenshotModal.component.jsx

```diff
@@ -8,17 +8,73 @@ import { Copy32Icon, Save24Icon } from '../../../icons';
 import { loginHandler } from '../SignInModal';
 import { useAuthContext } from '../../../contexts';
 import Toolbar from '../../scenegraph/Toolbar';
+import { db, storage } from '../../../services/firebase';
+import html2canvas from 'html2canvas';
+import { getDownloadURL, ref, uploadBytes } from 'firebase/storage';
+import {
+  collection,
+  doc,
+  getDoc,
+  serverTimestamp,
+  updateDoc
+} from 'firebase/firestore';
 
 function ScreenshotModal({ isOpen, onClose }) {
   const storedScreenshot = localStorage.getItem('screenshot');
   const parsedScreenshot = JSON.parse(storedScreenshot);
   const { currentUser } = useAuthContext();
-  const saveScreenshot = (value) => {
+  const saveScreenshot = async (value) => {
     const screenshotEl = document.getElementById('screenshot');
     screenshotEl.play();
     screenshotEl.setAttribute('screentock', 'type', value);
     screenshotEl.setAttribute('screentock', 'takeScreenshot', true);
+    try {
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+      const aframeScene = document.getElementById('screentock-destination');
```

Comment:
`aframeScene` is actually an img HTML element, can we rename to a more accurate ID for readibility such as `screentockImgElement`

(note to self: screentock-destination is created by toolbar component as img element that receives canvas drawing)